GVRf/Framework/src/org/gearvrf/FutureWrapper.java

```diff
@@ -0,0 +1,50 @@
+package org.gearvrf;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This is a wrapper to convert T to Future<T> so that we do not need many
+ * versions of constructor for {@link GVRSceneObject} for different combinations
+ * of {@link GVRMesh} or Future<{@link GVRMesh}>, {@link GVRTexture} or Future<
+ * {@link GVRTexture}>.
+ * 
+ * @param <V>
+ *            Internal type to be wrapped. It can be either mesh or texture.
+ */
+public class FutureWrapper<V> implements Future<V> {
+
+    private V value_ = null;
+
+    public FutureWrapper(V value) {
+        value_ = value;
+    }
+
+    @Override
+    public boolean cancel(boolean mayInterruptIfRunning) {
+        return false;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return false;
+    }
+
```

Comment:
Since this is a "faked" Future type, we cannot really cancel it. So "isCancelled" should always return "false" and "cancel" should do nothing but return "false".

For "isDone", since it is always "done" from the creating time, I have modified it to have it always return "true".